### PR TITLE
Implemented Series.compare

### PR DESCRIPTION
**Pull request by itholic**
This PR proposes `Series.compare`.

```python
>>> s1 = ks.Series(["a", "b", "c", "d", "e"])
>>> s2 = ks.Series(["a", "a", "c", "b", "e"])

>>> s1.compare(s2).sort_index()
  self other
1    b     a
3    d     b

>>> s1.compare(s2, keep_shape=True).sort_index()
  self other
0  NaN   NaN
1    b     a
2  NaN   NaN
3    d     b
4  NaN   NaN

>>> s1.compare(s2, keep_shape=True, keep_equal=True).sort_index()
  self other
0    a     a
1    b     a
2    c     c
3    d     b
4    e     e
```

### CODE REVIEW

File: databricks/koalas/tests/test_series.py [hunk 1]
```diff
@@ -2226,3 +2227,177 @@ def test_backfill(self):
             # Test `inplace=True`
             kser.backfill(inplace=True)
             self.assert_eq(expected, kser)
+
+    # TODO: Remove skip after dropping Python 3.5
+    @unittest.skipIf(
+        sys.version_info.minor <= 5,
```
Comment (HyukjinKwon) [inline databricks/koalas/tests/test_series.py:2233]: I think you should also compare major version.

File: databricks/koalas/tests/test_series.py [hunk 2]
```diff
@@ -2226,3 +2227,177 @@ def test_backfill(self):
...
+        "The columns order is not preserved in the result in Python<=3.5",
```
Comment (ueshin) [inline databricks/koalas/tests/test_series.py:2234]: Why is the result column order not preserved in Python 3.5?

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -5388,6 +5388,103 @@ def argmin(self):
...
+        Matching NaNs will not appear as a difference.
+
+        Examples
+        --------
+        >>> s1 = ks.Series(["a", "b", "c", "d", "e"])
+        >>> s2 = ks.Series(["a", "a", "c", "b", "e"])
+
+        Align the differences on columns
+
+        >>> s1.compare(s2).sort_index().sort_index()  # doctest: +SKIP
```
Comment (ueshin) [inline databricks/koalas/series.py:5422]: Why skipped?

File: databricks/koalas/series.py [hunk 4]
```diff
@@ -5388,6 +5388,103 @@ def argmin(self):
...
+          self other
+        1    b     a
+        3    d     b
+
+        Keep all original rows
+
+        >>> s1.compare(s2, keep_shape=True).sort_index()  # doctest: +SKIP
```
Comment (ueshin) [inline databricks/koalas/series.py:5429]: ditto.

File: databricks/koalas/series.py [hunk 5]
```diff
@@ -5388,6 +5388,103 @@ def argmin(self):
...
+           self other
+        0  None  None
+        1     b     a
+        2  None  None
+        3     d     b
+        4  None  None
+
+        Keep all original rows and also all original values
+
+        >>> s1.compare(s2, keep_shape=True, keep_equal=True).sort_index()  # doctest: +SKIP
```
Comment (ueshin) [inline databricks/koalas/series.py:5439]: ditto.

File: databricks/koalas/series.py (hunk 3 above)
Comment (ueshin) [reply databricks/koalas/series.py:5422]: Btw, please remove one of `.sort_index()`s. :)

Comment (itholic) [reply databricks/koalas/series.py:5422]: Sorry, my mistakes  😅

Thanks for the point out!

File: databricks/koalas/tests/test_series.py (hunk 2 above)
Comment (itholic) [reply databricks/koalas/tests/test_series.py]: Oh, It was test problem with `OrderedDict`.

Fixed test, thanks.

File: databricks/koalas/series.py (hunk 3 above)
Comment (ueshin) [reply databricks/koalas/series.py:5422]: Seems like these are not fixed yet?

File: databricks/koalas/series.py [hunk 6]
```diff
@@ -5388,6 +5388,103 @@ def argmin(self):
             scol_for(sdf, self._internal.data_spark_column_names[0]) == min_value[0]
         ).head()[0]
 
+    # TODO: Remove skip after dropping Python 3.5
+    def compare(self, other, keep_shape=False, keep_equal=False):
```
Comment (ueshin) [inline databricks/koalas/series.py:5392]: Could you add the type annotations?

File: databricks/koalas/series.py [hunk 7]
```diff
@@ -5388,6 +5388,103 @@ def argmin(self):
             scol_for(sdf, self._internal.data_spark_column_names[0]) == min_value[0]
         ).head()[0]
 
+    # TODO: Remove skip after dropping Python 3.5
```
Comment (ueshin) [inline databricks/koalas/series.py:5391]: I'm still not sure why we should skip with py3.5?

Comment (itholic) [reply databricks/koalas/series.py:5391]: Sorry we don't need to skip Python3.5.

I'll remove this comment, too.

File: databricks/koalas/series.py (hunk 6 above)
Comment (itholic) [reply databricks/koalas/series.py:5392]: Oh, yeah sure. Thanks!

File: databricks/koalas/series.py (hunk 7 above)
Comment (ueshin) [reply databricks/koalas/series.py]: Got it.

File: databricks/koalas/series.py [hunk 8]
```diff
@@ -5388,6 +5388,104 @@ def argmin(self):
...
+        keep_shape : bool, default False
+            If true, all rows and columns are kept.
+            Otherwise, only the ones with different values are kept.
+        keep_equal : bool, default False
+            If true, the result keeps values that are equal.
+            Otherwise, equal values are shown as NaNs.
+
+        Returns
+        -------
+        Series
```
Comment (ueshin) [inline databricks/koalas/series.py:5410]: nit: `DataFrame`?

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM.

File: databricks/koalas/series.py (hunk 8 above)
Comment (itholic) [reply databricks/koalas/series.py]: Thanks!

File: databricks/koalas/tests/test_series.py [hunk 9]
```diff
@@ -2226,3 +2226,178 @@ def test_backfill(self):
...
+    def test_compare(self):
+        if LooseVersion(pd.__version__) >= LooseVersion("1.1"):
+            pser1 = pd.Series(["b", "c", np.nan, "g", np.nan])
+            pser2 = pd.Series(["a", "c", np.nan, np.nan, "h"])
+            kser1 = ks.from_pandas(pser1)
+            kser2 = ks.from_pandas(pser2)
+            self.assert_eq(
+                pser1.compare(pser2).sort_index(),
+                kser1.compare(kser2).sort_index(),
+                check_exact=False,
```
Comment (ueshin) [inline databricks/koalas/tests/test_series.py:2239]: Do we need this and the followings?
Seems like it passes without it.

Comment (itholic) [reply databricks/koalas/tests/test_series.py:2239]: Oh, yeah we don't need it.

Thanks :)

**Comment by ueshin** [comment]
Thanks! merging.

**Comment by itholic** [comment]
Thanks, @ueshin :)

**Pull request merged**